src/test/harness.rs

```diff
@@ -27,20 +27,57 @@ use rls_analysis::{AnalysisHost, Target};
 use rls_vfs::Vfs;
 use serde_json;
 
+#[path = "../../tests/support/mod.rs"]
```

Comment:
Would you mind if we leave it for now?

If we factor out the file/project creation facilities, we'd still depend on [paths.rs](https://github.com/rust-lang/rls/blob/master/tests/support/paths.rs), which I'd say doesn't make sense to publish as a crate. Personally, I'd avoid converting the project layout to workspace just to accommodate for that helper crate, since unifying those tests later on will mean we stop depending on this module in `src/` dir, and so we won't need this as a standalone crate anymore.

We could experiment with creating those scratchpad environments using temporary directories instead of using `paths.rs` (generated under `target/` with TLS counter) but it works for Cargo and wasn't sure if creating/running stuff inside temporary dirs would work on Rust CI.
@kennytm @pietroalbini Could the pattern of: create some `/tmp` writable dir -> copy test files over -> run RLS there pose any problems from Rust CI perspective?

Reply:
Done; Now `src/test/` tests generate scratchpad environment using temp dirs from the `tempfile` crate and we don't need to `#[path = "..."]` anything anymore.